Refresh query docs. (#9704)
apache/druid: docs/querying/joins.md

  ~ under the License.
  -->

Druid has two features related to joining of data:

1. [Join](datasource.md#join) operators. These are available using a [join datasource](datasource.md#join) in native
queries, or using the [JOIN operator](sql.md#query-syntax) in Druid SQL. Refer to the
[join datasource](datasource.md#join) documentation for information about how joins work in Druid.
2. [Query-time lookups](lookups.md), simple key-to-value mappings. These are preloaded on all servers and can be
accessed with or without an explicit join operator. Refer to the [lookups](lookups.md) documentation for more details.

> Whenever possible, for best performance it is good to avoid joins at query time. Often this can be accomplished by

Reviewer: Slightly reworded but should it say "avoid" or just "use wisely"?

```suggestion
For best performance, we recommend avoiding joins at query time whenever possible. Often this can be accomplished by
```

> joining data before it is loaded into Druid.

Author: I think those examples are really the same thing. I combined them added another. Here is the result. How does it look?

> Whenever possible, for best performance it is good to avoid joins at query time. Often this can be accomplished by joining data before it is loaded into Druid. However, there are situations where joins or lookups are the best solution available despite the performance overhead, including:
>
> - The fact-to-dimension (star and snowflake schema) case: you need to change dimension values after initial ingestion, and aren't able to reingest to do this. In this case, you can use lookups for your dimension tables.
> - Your workload requires joins or filters on subqueries.

> joining data before it is loaded into Druid.

Reviewer 2: looks good. It appears I can't resolve these comments so I've just been emoji-ing the ones I agree with so that I don't spam you with notifications
